frontend/packages/dev-console/src/components/topology/Topology.tsx

```diff
@@ -173,6 +176,24 @@ const Topology: React.FC<ComponentProps> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [data]);
 
+  React.useEffect(() => {
+    if (!displayFiltersRef.current) {
+      displayFiltersRef.current = filters.display;
+      return;
+    }
+
+    if (
+      (filters.display.eventSources && !displayFiltersRef.current.eventSources) ||
+      (filters.display.virtualMachines && !displayFiltersRef.current.virtualMachines)
+    ) {
+      action(() => {
+        visRef.current.getGraph().reset();
+        visRef.current.getGraph().layout();
+      })();
+    }
+    displayFiltersRef.current = filters.display;
+  }, [filters.display]);
```

Comment:
This code is too specific when what we really want is to run layout on nodes that change visibility.
As discussed, an event to notify of visibility change to let the layout handle this much like add / remove of nodes.

Reply:
Also discussed doing such changes in a future issue.

Reply:
https://issues.redhat.com/browse/ODC-3616